src/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs

```diff
@@ -114,5 +114,31 @@ public void WriteToReadOnlyFile_UnauthException()
             Assert.Throws<UnauthorizedAccessException>(() => File.WriteAllBytes(path, Encoding.UTF8.GetBytes("text")));
             File.SetAttributes(path, FileAttributes.Normal);
         }
+
+        /// <summary>
+        /// On Unix, modifying a file that is ReadOnly will fail under normal permissions.
+        /// If the test is being run under the superuser, however, modification of a ReadOnly
+        /// file is allowed. Since we don't know the user running these tests, we attempt
+        /// the operation and verify the result for either case.
+        /// </summary>
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void Unix_WriteToReadOnlyFile()
+        {
+            string path = GetTestFilePath();
+            File.Create(path).Dispose();
+            File.SetAttributes(path, FileAttributes.ReadOnly);
+
+            try
+            {
+                File.WriteAllBytes(path, Encoding.UTF8.GetBytes("text"));
+                Assert.Equal(Encoding.UTF8.GetBytes("text"), File.ReadAllBytes(path));
+            }
```

Comment:
Is there some kind of assert we could add here (and in other similar places) that we're running in an elevated context and that's why it didn't throw? e.g. would it work to check whether geteuid() returns 0?